provider/core/src/request.rs

```diff
@@ -246,6 +249,22 @@ impl FromStr for DataLocale {
     }
 }
 
+impl PartialOrd for DataLocale {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for DataLocale {
+    fn cmp(&self, other: &Self) -> Ordering {
+        self.langid
+            .as_tuple()
```

Comment:
Note to self: we shouldn't implement this, even on DataLocale, due to the same reasons as discussed in https://github.com/unicode-org/icu4x/issues/1215. We should instead implement `Ord` only on the private `ResolvedLocaleInfo` type.

Reply:
https://github.com/unicode-org/icu4x/pull/4608